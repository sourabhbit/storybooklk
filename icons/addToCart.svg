<svg width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.2468 11.7257H7.35978L4.48144 0.239258H0.32666V1.65844H3.37161L6.24996 13.1449H15.1926L18.9731 4.09277L17.6608 3.54557L14.2468 11.7257Z" fill="black"/>
<path d="M6.80349 18.1798C7.59903 18.1798 8.24394 17.5364 8.24394 16.7427C8.24394 15.9491 7.59903 15.3057 6.80349 15.3057C6.00795 15.3057 5.36304 15.9491 5.36304 16.7427C5.36304 17.5364 6.00795 18.1798 6.80349 18.1798Z" fill="black"/>
<path d="M14.0008 18.1798C14.7963 18.1798 15.4412 17.5364 15.4412 16.7427C15.4412 15.9491 14.7963 15.3057 14.0008 15.3057C13.2052 15.3057 12.5603 15.9491 12.5603 16.7427C12.5603 17.5364 13.2052 18.1798 14.0008 18.1798Z" fill="black"/>
<path d="M11.8324 7.4095V4.88822H14.3596V3.46905H11.8324V0.947754H10.4099V3.46905H7.88263V4.88822H10.4099V7.4095H11.8324Z" fill="black"/>
</svg>
