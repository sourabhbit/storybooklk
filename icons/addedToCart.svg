<svg width="20" height="19" viewBox="0 0 20 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M14.461 11.7257H7.57658L4.69823 0.239258H0.543457V1.65844H3.58584L6.46421 13.1449H15.4094L19.1899 4.09277L17.8776 3.54557L14.461 11.7257Z" fill="#000042"/>
<path d="M8.44037 16.8823C8.51873 16.0925 7.9405 15.3889 7.14884 15.3107C6.35718 15.2325 5.65189 15.8094 5.57353 16.5992C5.49517 17.389 6.0734 18.0927 6.86506 18.1709C7.65672 18.249 8.36201 17.6721 8.44037 16.8823Z" fill="#000042"/>
<path d="M15.6362 16.8814C15.7146 16.0916 15.1363 15.388 14.3447 15.3098C13.5531 15.2316 12.8478 15.8085 12.7694 16.5983C12.6911 17.3881 13.2693 18.0917 14.0609 18.1699C14.8526 18.248 15.5578 17.6712 15.6362 16.8814Z" fill="#000042"/>
<path d="M14.2996 1.00391L10.8856 4.97249L8.92479 3.28226L7.99438 4.35624L11.0342 6.97469L15.3786 1.92701L14.2996 1.00391Z" fill="#000042"/>
</svg>
