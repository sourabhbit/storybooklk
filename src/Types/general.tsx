export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
