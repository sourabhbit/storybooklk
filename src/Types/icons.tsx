export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
