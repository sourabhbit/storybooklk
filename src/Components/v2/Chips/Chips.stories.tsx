import { ComponentMeta, ComponentStory } from '@storybook/react'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import { generateArgTypes, generateGeneralArgs } from 'Helpers/Storybook'
import Chip from './index'
import { iconsENUM } from 'Types/icons'

export default {
	title: 'Components/v2/Chips',
	component: Chip,
	argTypes: {
		...generateGeneralArgs(),
		text: generateArgTypes({
			type: 'string',
			required: true,
			description: 'Text that will be rendered',
			category: 'Component',
		}),
		showIcon: generateArgTypes({
			type: 'boolean',
			required: true,
			description: 'Boolean to show/hide icon',
			category: 'Component',
			defaultValue: 'false',
		}),
		icon: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Select Icon',
			category: 'Component',
			control: 'select',
			defaultValue: 'glasses',
			options: iconsENUM,
		}),
		count: generateArgTypes({
			type: 'number',
			required: false,
			description: 'Show/Hide Count',
			category: 'Component',
		}),
		color: generateArgTypes({
			type: 'number',
			required: false,
			description: 'Show Count',
			category: 'Component',
		}),
		showColor: generateArgTypes({
			type: 'number',
			required: false,
			description: 'Show/Hide color',
			category: 'Component',
		}),
	},
} as ComponentMeta<typeof Chip>

const Template: ComponentStory<typeof Chip> = (args) => <Chip {...args} />

export const Active = Template.bind({})
Active.args = {
	id: 'active-id',
	text: 'Active Chip',
	count: 10,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.primary,
	font: TypographyENUM.serif,
	showIcon: false,
	icon: iconsENUM.glasses,
}

export const Inactive = Template.bind({})
Inactive.args = {
	id: 'inactive-id',
	text: 'In Active Chip',
	count: 12,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	showIcon: false,
	icon: iconsENUM.glasses,
	color: 'green',
	showColor: true,
}

export const InactiveWithIcon = Template.bind({})
InactiveWithIcon.args = {
	id: 'inactive-id',
	text: 'In Active Chip',
	count: 12,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	showIcon: true,
	icon: iconsENUM.glasses,
}
