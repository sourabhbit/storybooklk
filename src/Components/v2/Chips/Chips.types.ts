import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
export interface ChipsTypes {
	id: string
	showIcon: boolean
	text: string
	color?: string
	showColor?: boolean
	theme: ThemeENUM
	font: TypographyENUM
	componentSize?: ComponentSizeENUM
	count?: number
	icon?: iconsENUM
}
