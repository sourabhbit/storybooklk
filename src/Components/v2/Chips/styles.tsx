import styled from 'styled-components'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'

const getBGColor = (colorType: ThemeENUM) => {
	if (colorType === ThemeENUM.primary) {
		return 'var(--dark-blue-100)'
	} else {
		return 'var(--white)'
	}
}

const getColor = (colorType: ThemeENUM, count: boolean) => {
	if (colorType === ThemeENUM.primary) {
		return count ? 'var(--see-green-25)' : 'var(--see-green-100)'
	} else {
		return count ? 'var(--dark-blue-50)' : 'var(--dark-blue-100)'
	}
}

const getFontSize = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return 'var(--fs-12)'
	} else if (size === ComponentSizeENUM.medium) {
		return 'var(--fs-18)'
	} else {
		return 'var(--fs-30)'
	}
}

const getMargin = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return 'var(--spacing-sm)'
	} else if (size === ComponentSizeENUM.medium) {
		return 'var(--spacing-md)'
	} else {
		return 'var(--spacing-lg)'
	}
}

const getHeight = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return '15px'
	} else if (size === ComponentSizeENUM.medium) {
		return '20px'
	} else {
		return '30px'
	}
}

const getWidth = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return '15px'
	} else if (size === ComponentSizeENUM.medium) {
		return '20px'
	} else {
		return '30px'
	}
}

export const ChipWrapper = styled.div<{
	styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: var(--spacing-sm) var(--spacing-md);
	border-radius: var(--border-radius);
	background-color: ${(props) => getBGColor(props.styleColor)};
	color: ${(props) => getColor(props.styleColor, false)};
	font-size: ${(props) => getFontSize(props.styleSize)};
	font-family: ${(props) => props.styleFont};
	max-width: fit-content;
`
export const CountWrapper = styled.span<{
	styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
}>`
	color: ${(props) => getColor(props.styleColor, true)};
	margin-left: ${(props) => getMargin(props.styleSize)};
`

export const ColorWrapper = styled.span<{
	styleSize: ComponentSizeENUM
}>`
	background: green;
	border-radius: 50%;
	width: ${(props) => getWidth(props.styleSize)};
	height: ${(props) => getHeight(props.styleSize)};
`

export const TextWrapper = styled.span<{
	styleSize: ComponentSizeENUM
}>`
	margin-left: ${(props) => getMargin(props.styleSize)};
`
