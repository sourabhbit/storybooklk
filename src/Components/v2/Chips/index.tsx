import { ChipWrapper, CountWrapper, TextWrapper, ColorWrapper } from './styles'
import { ChipsTypes } from './Chips.types'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import SVGIcon from 'Components/common/SVGIcon'

/**
 * Default Chips Component
 */
const Chip = ({
	id,
	showIcon = false,
	text,
	count = 0,
	icon,
	color = '',
	showColor = false,
	componentSize = ComponentSizeENUM.small,
	theme = ThemeENUM.primary,
	font = TypographyENUM.serif,
}: ChipsTypes) => {
	const showCount = count !== 0 ? true : false
	return (
		<ChipWrapper
			data-testid="Chip-wrapper"
			id={id}
			styleFont={font}
			styleSize={componentSize}
			styleColor={theme}
		>
			{showIcon && icon && <SVGIcon icon={icon} alt="test" />}
			{showColor && color && <ColorWrapper styleSize={componentSize} />}
			<TextWrapper styleSize={componentSize} data-testid="Chip-text">
				{text}
			</TextWrapper>
			{showCount && (
				<CountWrapper
					styleColor={theme}
					styleSize={componentSize}
					data-testid="Chip-count"
				>
					{count}
				</CountWrapper>
			)}
		</ChipWrapper>
	)
}

export default Chip
