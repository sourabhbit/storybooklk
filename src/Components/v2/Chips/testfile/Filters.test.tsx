import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
import Chips from '../index'

const props = {
	id: 'active-id',
	text: 'Active Chips',
	count: 10,
	color: 'green',
	showColor: true,
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.primary,
	font: TypographyENUM.serif,
	showIcon: false,
	icon: iconsENUM.glasses,
	onClick: jest.fn(),
}

describe('<Chips />', () => {
	let _asFragment: any = null
	let _rerender: any = null

	beforeEach(() => {
		const { asFragment, rerender } = render(<Chips {...props} />)
		_asFragment = asFragment
		_rerender = rerender
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('Chip-wrapper')
		const count = screen.getByTestId('Chip-count')
		const text = screen.getByTestId('Chip-text')
		// const color = screen.getByTestId('Chip-color')
		expect(container).toBeInTheDocument()
		expect(count).toBeInTheDocument()
		expect(text).toBeInTheDocument()
		// expect(color).toBeInTheDocument()
	})

	it('Should not render component', () => {
		_rerender(<Chips {...props} count={undefined} />)
		const count = screen.queryByTestId('Chip-count')
		const color = screen.queryByTestId('Chip-color')
		expect(count).not.toBeInTheDocument()
		expect(color).not.toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
