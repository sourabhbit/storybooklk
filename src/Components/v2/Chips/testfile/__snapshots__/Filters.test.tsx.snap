// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chips /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kffJJs"
    data-testid="Chip-wrapper"
    id="active-id"
  >
    <span
      class="sc-dkPtRN gRmiDn"
    />
    <span
      class="sc-hKwDye gRuEna"
      data-testid="Chip-text"
    >
      Active Chips
    </span>
    <span
      class="sc-gsDKAQ eSbkki"
      data-testid="Chip-count"
    >
      10
    </span>
  </div>
</DocumentFragment>
`;
